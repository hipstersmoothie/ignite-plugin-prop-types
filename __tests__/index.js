import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

import PropTypesComponent from '../src';

const testPlugin = () => {};

testPlugin.propTypes = {
  children: PropTypes.node,
  string: PropTypes.string.isRequired,
  number: PropTypes.number
};

testPlugin.defaultProps = {
  children: null,
  string: PropTypes.string.isRequired,
  number: 10
};

const testPlugins = {
  test: {
    provider: () => Promise.resolve({ default: testPlugin })
  }
};

describe('PropTypesComponent', () => {
  test('renders nothing when no component found', done => {
    const component = renderer.create(
      <PropTypesComponent component="notfound" plugins={testPlugins} />
    );

    setImmediate(() => {
      expect(component.toJSON()).toBeNull();
      done();
    });
  });

  test('finds component provider', done => {
    const component = renderer.create(
      <PropTypesComponent component="test" plugins={testPlugins} />
    );

    setImmediate(() => {
      expect(component.toJSON()).toMatchSnapshot();
      done();
    });
  });
});
