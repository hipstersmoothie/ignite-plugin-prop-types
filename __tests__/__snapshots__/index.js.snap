// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropTypesComponent finds component provider 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Property
      </th>
      <th>
        Type
      </th>
      <th>
        Default
      </th>
      <th>
        Required
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        children
      </td>
      <td>
        node
      </td>
      <td>
        null
      </td>
      <td>
        false
      </td>
    </tr>
    <tr>
      <td>
        string
      </td>
      <td>
        string
      </td>
      <td>
        func
      </td>
      <td>
        true
      </td>
    </tr>
    <tr>
      <td>
        number
      </td>
      <td>
        number
      </td>
      <td>
        10
      </td>
      <td>
        false
      </td>
    </tr>
  </tbody>
</table>
`;
